[teamcity] Add ssl_validation flag to check

jslatts opened the pull request:
It would be helpful to have the ability to disable SSL validation in the teamcity check, same idea as issue #1782.

I went ahead and added a flag in with the same name as the nginx check. I did not try and write an integration test. Is a test needed for this?

remh commented:
Thanks @jslatts 
A test is not needed in that case as it would basically be testing the requests library.

We'll review more in details for our 5.7.0 release.
Thanks again!

remh commented on checks.d/teamcity.py:
```diff
@@ -21,7 +21,7 @@ def __init__(self, name, init_config, agentConfig, instances=None):
         self.last_build_ids = {}
 
-    def _initialize_if_required(self, instance_name, server, build_conf):
+    def _initialize_if_required(self, instance_name, server, build_conf, ssl_validation):
```
Looks like you forgot to pass the ssl_validation flag when calling `_initialize_if_required`

irabinovitch replied:
@jslatts The window for the 5.7  release will be closing soon. Do you have a moment to look at @remh 's feedback?

jslatts replied:
@irabinovitch sorry, missed the original notification. I added it in. Do you want me to rebase the PR against current master?

olivielpeau commented on conf.d/teamcity.yaml.example:
```diff
@@ -22,6 +22,9 @@ instances:
     # rather than just that a successful build happened
     # is_deployment: true
 
+    # Optional, this turns of ssl ceritificate validation. Defaults to True.
```
Small typos here: `turns of ssl ceritificate validation` -> `turns off ssl certificate validation`

olivielpeau commented on checks.d/teamcity.py:
```diff
@@ -91,6 +91,8 @@ def check(self, instance):
         if instance_name is None:
             raise Exception("Each instance must have a unique name")
 
+        ssl_validation = instance.get('ssl_validation', True)
```
Could you use the `_is_affirmative` function here? i.e: 

```python
ssl_validation = _is_affirmative(instance.get('ssl_validation', True))
```

olivielpeau commented:
Hi @jslatts!

I've added two comments, could you address them? Also, could you only keep your commits in this PR? The other 3 commits seem to be unrelated.

Apart from that your changes look good to me. Thanks!

jslatts commented:
Fixed the typo and added the _is_affirmative call. Also rebased against current master. No idea how the other commits ended up in there. It looks clean now. Let me know if there is anything else.

olivielpeau commented:
Thanks again @jshirley!

Merging.
